airflow_dbt_python/hooks/dbt.py

```diff
@@ -555,22 +599,101 @@ class DbtSnowflakeHook(DbtHook):
     conn_type = "snowflake"
     hook_name = "dbt Snowflake Hook"
     conn_params = [
-        DbtConnectionParam("conn_type", "type", "postgres"),
+        DbtConnectionParam("conn_type", "type", conn_type),
         "host",
         "schema",
-        DbtConnectionParam("login", "user"),
-        "password",
+        DbtConnectionParam(
+            "login",
+            "user",
+            depends_on=lambda x: x.extra_dejson.get("authenticator", "") != "oauth",
+        ),
+        DbtConnectionParam(
+            "login",
+            "oauth_client_id",
+            depends_on=lambda x: x.extra_dejson.get("authenticator", "") == "oauth",
+        ),
+        DbtConnectionParam(
+            "password",
+            depends_on=lambda x: not any(
+                (
+                    *(k in x.extra_dejson for k in ("private_key_file", "private_key_content")),
+                    x.extra_dejson.get("authenticator", "") == "oauth",
+                ),
+            ),
+        ),
+        DbtConnectionParam(
+            "password",
+            "private_key_passphrase",
+            depends_on=lambda x: any(
+                k in x.extra_dejson for k in ("private_key_file", "private_key_content")
```

Comment:
Shouldn't we check for `x.extra_dejson.get("authenticator", "") != "oauth"` to be consistent with other parameters?

Reply:
It's also less mental load for the user, from the user's PoV it's easier to reason about the state of a single key.

Reply:
Closing this one as it's the same as the previous comment.